File: Problems 0700-0799/769. Max Chunks To Make Sorted/solution.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int maxPos = 0, numChunks = 0;
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] > maxPos) {
                maxPos = arr[i];
            }
            if (maxPos == i) numChunks++;
        }
        return numChunks;
    }
};

File: Problems 0700-0799/769. Max Chunks To Make Sorted/solution.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        currEnd = -1
        res = 0
        for i, num in enumerate(arr):
            currEnd = max(num, currEnd)
            if i == currEnd:
                res += 1
        return res